### DISCUSSION TITLE
Violin Plots

### DISCUSSION THREAD

**Discussion by Usha-88** [satijalab/seurat#7797] [General]
![image](https://github.com/satijalab/seurat/assets/145045016/b72c208c-fa48-4b15-b01a-3622a5012ce6)

In this particular image,

what do the dots represent and what does the violin plot resemble? Since the Y-axis is termed as expression level, why cant we see a violin plot for 0,1,2 clusters even though there is some expression represented by the dots.

Thank you.

**Comment by samuel-marsh** [comment]
For more on violin plots you can find many resources easily via Google but here is paper cited by Seurat's function which describes it as well:
https://www.stat.cmu.edu/~rnugent/PCMI2016/papers/ViolinPlots.pdf
Another resource here: https://mode.com/blog/violin-plot-examples/

Each dot is a cell and the height of dot on y axis is the normalized expression level.  The shape of the violin is based on the distribution/density of the data.  In cluster 0 for instance there are only 4 cells which express the gene.  The overwhelming majority of cells have 0 expression.  Therefore the violin has no visible shape.

Best,
Sam

**Reply by Usha-88** [reply]
![image](https://github.com/satijalab/seurat/assets/145045016/8163f42d-4476-437a-932b-96b184591103)
Does that mean, the multiple dots in clusters 0,1,2,3 have 0 expression of NKG7? Then why do the dots appear in these clusters?

**Comment by samuel-marsh** [comment]
Hi,
The dots that align with zero on the y-axis have zero expression yes.  You can see this clearly in cluster 6 in which all cells have some level of expression of NKG7 and therefore there are no dots at zero.

Best,
Sam

**Reply by Usha-88** [reply]
Thank you.
